### PR TITLE
Wordpress comic ripper now supports more sites

### PR DESCRIPTION
**Pull request by cyian-1756**
Test links

http://www.totempole666.com/comic/first-time-for-everything-00-cover/

http://buttsmithy.com/archives/comic/p1

http://themonsterunderthebed.net/?comic=test-post

http://prismblush.com/comic/hella-trap-pg-01/

http://www.konradokonski.com/sawdust/

http://www.konradokonski.com/wiory/

http://freeadultcomix.com/finders-feepaid-in-full-sparrow/

http://comics-xxx.com/republic-rendezvous-palcomix-star-wars-xxx/

### CODE REVIEW

File: src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java [hunk 1]
```diff
@@ -94,6 +131,36 @@ public String getAlbumTitle(URL url) throws MalformedURLException {
...
+                return "konradokonski.com_wiory";
+            }
+
+            Pattern freeadultcomixPat = Pattern.compile("https?://freeadultcomix.com/([a-zA-Z0-9_\\-]*)/?$");
+            Matcher freeadultcomixMat = freeadultcomixPat.matcher(url.toExternalForm());
+            if (freeadultcomixMat.matches()) {
+                return getHost() + "_" + freeadultcomixMat.group(1);
+            }
+
+            Pattern thisisDelvecomicPat = Pattern.compile("https?://thisis.delvecomic.com/NewWP/comic/([a-zA-Z0-9_\\-]*)/?$");
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java]: Is `NewWP` the only possible value for that part of this URL?

File: src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java [hunk 2]
```diff
@@ -119,10 +186,12 @@ public Document getNextPage(Document doc) throws IOException {
             // Find next page
             String nextPage = "";
             Element elem = null;
-            if (explicit_domains.contains("www.totempole666.com") == true
-            || explicit_domains.contains("buttsmithy.com") == true
-            || explicit_domains.contains("themonsterunderthebed.net")
-            || explicit_domains.contains("prismblush.com")) {
+            if (getHost().contains("www.totempole666.com") == true
+            || getHost().contains("buttsmithy.com") == true
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java]: > `== true`

While you're here could you clean those up?

File: src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java [hunk 3]
```diff
@@ -133,31 +202,78 @@ public Document getNextPage(Document doc) throws IOException {
...
                     elem = doc.select("div.comic-table > div#comic > img").first();
                 }
+                // Check if this is a site where we can get the page number from the title
+                if (url.toExternalForm().contains("buttsmithy.com") == true) {
+                    // Set the page title
+                    pageTitle = doc.select("meta[property=og:title]").attr("content");
+                    pageTitle = pageTitle.replace(" ", "");
+                    pageTitle = pageTitle.replace("P", "p");
+                }
+                if (url.toExternalForm().contains("www.totempole666.com") == true) {
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java]: Ditto for all of these.... `if (b == true)` is the same as `if (b)`

File: src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java [hunk 4]
```diff
@@ -133,31 +202,78 @@ public Document getNextPage(Document doc) throws IOException {
...
             }
 
         @Override
         public List<String> getURLsFromPage(Document doc) {
             List<String> result = new ArrayList<String>();
-            if (explicit_domains.contains("www.totempole666.com") == true
-            || explicit_domains.contains("buttsmithy.com") == true
-            || explicit_domains.contains("themonsterunderthebed.net")
-            || explicit_domains.contains("prismblush.com")) {
+            if (getHost().contains("www.totempole666.com") == true
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java]: `a == true` has the same boolean value as `a`

Please remove the redundant `== true` checks.

File: src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java [hunk 5]
```diff
@@ -133,31 +202,78 @@ public Document getNextPage(Document doc) throws IOException {
...
+                for (Element elem : doc.select("div.single-post > center > p > img")) {
+                    result.add(elem.attr("src"));
+                }
+            }
             return result;
         }
 
         @Override
         public void downloadURL(URL url, int index) {
+            sleep(500);
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java]: Is there any rate limiting in play that would make a delay useful? (If not, we're just slowing down the rip for no reason.)

File: src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java (hunk 1 above)
Comment (cyian-1756) [inline src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java]: Yes it is.

File: src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java (hunk 5 above)
Comment (cyian-1756) [inline src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java]: There doesn't seem to be any rate limiting

File: src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java (hunk 3 above)
Comment (cyian-1756) [inline src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java]: fixed in my last push

File: src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java (hunk 2 above)
Comment (cyian-1756) [inline src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java]: fixed in my last push

File: src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java (hunk 4 above)
Comment (cyian-1756) [inline src/main/java/com/rarchives/ripme/ripper/rippers/WordpressComicRipper.java]: fixed in my last push

**Comment by cyian-1756** [comment]
@metaprime Can I get an update on when this will be merged?

**Comment by metaprime** [comment]
Sorry, I've been really busy with work. Going out of town for the long weekend. Will merge probably sometime next week.

**Pull request merged**